### ISSUE TITLE
Effect of f0 and energy Dropout in FastSpeech2

### ISSUE THREAD

**Issue by MMingabc** [TensorSpeech/TensorFlowTTS#529]
Hi,
I am curious about the effect of f0 and energy dropout (in both the variant predictor and f0/energy embedding) in FastSpeech2 . The default dropout rate is 0.5 which is quite big and we have to turn it on during inference. What's the point? I tested the results with dropout off during inference. The sound is pale without any color. Why? What if we turn the ratio down to like 0.1 and turn it off during inference. Is the result gonna be better or worse? 
I trained a fastspeech and an mb-melgan for CN TTS. The overall result is fastifying but I am experiencing an unstable performance. For the same sentence, sometime the speech is perfect but sometimes there are artefacts. I am wondering whether this results from the random dropout. 
Hope someone could explain this to me. Thank you very much!

**Comment by dathudeptrai** [comment]
@MMingabc large dropout for f0 and energy aims to reduce the contribution of f0/energy to the mel-reconstructed. In other words, it reduces the model overfitted.

**Comment by MMingabc** [comment]
> @MMingabc large dropout for f0 and energy aims to reduce the contribution of f0/energy to the mel-reconstructed. In other words, it reduces the model overfitted.

@dathudeptrai Thanks for your reply! My FastSpeech is not very stable now. Sometimes there are artefacts especially at ends of sentences. What should I do? Looking forward to your advices,

**Comment by dathudeptrai** [comment]
> > @MMingabc large dropout for f0 and energy aims to reduce the contribution of f0/energy to the mel-reconstructed. In other words, it reduces the model overfitted.
> 
> @dathudeptrai Thanks for your reply! My FastSpeech is not very stable now. Sometimes there are artefacts especially at ends of sentences. What should I do? Looking forward to your advices,

the simple solution for a problem with the end of sentences is that you add some words in the end and then erase it by a predicted duration :D.
